src/OrbitGl/CaptureViewElement.cpp

```diff
@@ -87,6 +110,31 @@ void CaptureViewElement::OnDrag(int x, int y) {
   RequestUpdate();
 }
 
+std::vector<CaptureViewElement*> CaptureViewElement::GetVisibleChildren() const {
```

Comment:
As above, I renamed this and the other "GetXXXChildren" methods. I think we need all three of them:

- `GetAllChildren` is required because we need to be able to iterate over all children and e.g. update visibility
- `GetNonHiddenChildren` is required to implement e.g. filtering (on top of explicitly setting a track to be "hidden"), and we want to be able to iterate over all children that should be rendered independent of the viewport. Currently, we iterate over all children that need to be rendered to determine scrolling offset, and only then can it actually be determined if an element is visible in the viewport.
- `GetChildrenVisibleInViewport` is mostly for performance optimization to not render off-screen elements, and will in the future be used for accessibility.

One way to get rid of `GetNonHiddenChildren`: `Track::ShouldBeRendered` could take the track type visibility of track manager into account. I didn't want this because it will result in an "upward" dependency from `Track` to `TrackManager`, but on the other hand it would not require `GetNonHiddenChildren` to be virtual because it will simple filter out all children with `ShouldBeRendered() == true`.

More consistently, I could introduce `Track::SetIsFiltered` and have the filtering use this property instead. This removes the method as well without introducing another dependency, but adds additional methods to the track API. WDYT?

(Disclaimer, I don't want to use `SetVisible` for this because I want to be able to explicitly show / hide single tracks independent of the track type visibility)

Reply:
1. Ok with the renaming, thanks for that.
2. I don't like it.
3. It could be.
4. About disclaimer. I feel Track visibility is not very clear (you have to go through several parts to understand all the complex logic of visibility) What do you think about creating TrackVisibility as a combination of boolean flags? You can have:
- A flag for Track visibility (currently not used)
- A flag of Type visibility (set by UpdateTypeVisibility())
- Maybe Filtering? (UpdateFilteringVisibility). This particular refactor could simplify a lot TrackManager's code, because TrackManager only need to run a method when the filter is modified.
- Maybe Emptyness? 
In that sense, all the logic related to visible Tracks will be only in one component (Track.h). If you don't like it, consider at least mix the first two (Track visibility and Type visibility).

Reply:
I like the idea of making visibility not a boolean, but a combination of flags. I'll look into this, thanks!

Reply:
As discussed offline, I will merge this as-is and look into reworking the visibility in a separate PR